src/core_modules/capture-core-utils/RulesEngine/RulesEngine.js

```diff
@@ -1,117 +1,379 @@
-        const dateUtils = getDateUtils(momentConverter);
-        const variableService = new VariableService(valueProcessor.processValue, dateUtils);
-        this.executionService = getExecutionService(onTranslate, variableService, dateUtils);
-        this.onProcessRulesEffects = getRulesEffectsProcessor(
-            this.executionService.convertDataToBaseOutputValue,
-            outputRulesConverterObject,
-        );
+    static dateUtils = getDateUtils(momentConverter)
+
+    static variableService() {
+        const valueProcessor = new ValueProcessor(inputValueConverter);
+        return new VariableService(valueProcessor.processValue);
     }
 
-    executeRules(
+
+    static programRuleEffectsForTEI(
         programRulesContainer: ProgramRulesContainer,
-        executingEvent: ?EventData | {},
-        eventsData: ?EventsData,
-        dataElements: ?DataElements,
         enrollmentData: ?Enrollment,
         teiValues: ?TEIValues,
         trackedEntityAttributes: ?TrackedEntityAttributes,
         selectedOrgUnit: OrgUnit,
-        optionSets: ?OptionSets,
-        processType: $Values<typeof processTypes>,
-    ): ?Array<OutputEffect> {
-        let eventsContainer;
-        if (eventsData && eventsData.length > 0) {
-            const eventsDataByStage = eventsData.reduce((accEventsByStage, event) => {
-                const hasProgramStage = !!event.programStageId;
-                if (hasProgramStage) {
-                    accEventsByStage[event.programStageId] = accEventsByStage[event.programStageId] || [];
-                    accEventsByStage[event.programStageId].push(event);
-                }
-                return accEventsByStage;
-            }, {});
-
-            eventsContainer = {
-                all: eventsData,
-                byStage: eventsDataByStage,
-            };
-        } else {
-            eventsContainer = null;
-        }
+        optionSets: OptionSets,
+    ): ?OutputEffects {
+        const { programRules } = programRulesContainer;
 
-        const effects = this.executionService.executeRules(
+        const variablesHash = RulesEngine.variableService().getVariables(
             programRulesContainer,
-            executingEvent,
-            eventsContainer,
-            dataElements,
+            null,
+            null,
+            null,
             trackedEntityAttributes,
             teiValues,
             enrollmentData,
             selectedOrgUnit,
             optionSets,
-            { debug: true },
         );
 
-        const processedEffects = effects ?
-            this.onProcessRulesEffects(effects, processType, dataElements, trackedEntityAttributes) :
-            null;
-        return processedEffects;
+        const dhisFunctions = d2Functions(RulesEngine.dateUtils, RulesEngine.variableService(), variablesHash);
+        return getProgramRuleEffects(dhisFunctions, programRules, null, trackedEntityAttributes, variablesHash, processTypes.TEI);
```

Comment:
Strange. I am only getting an eslint warning which says the line is more than 120 characters. 

I am happy to fix that but is that what you are talking as well? It might be that my flow is still not set up properly (?)